### PR TITLE
fixed some bugs, add more types.

### PR DESCRIPTION
**Pull request by wangjwchn**
This PR made the following modifications:
 - Fix the bug that the checker will crash when there is a `'` in the variable name.
 - Display the input of Z3 for debugging
 - Add more types:
    - nat
    - Z
    - bool
    - int64
Simples can be found in the file `test-suite/Test.v`. (Some operations of them have not been translated yet, the work is in progress now)

### CODE REVIEW

File: _CoqProject [hunk 1]
```diff
@@ -1,5 +1,5 @@
 -I src
--Q theories SMTC
+-R theories SMTC
```
Comment (gmalecha) [inline _CoqProject:2]: `-Q` is a much better way to use libraries than `-R`.

File: _CoqProject [hunk 2]
```diff
@@ -11,4 +11,6 @@ src/cvc4Solver.ml
 src/polyaSolver.ml
 src/smtTactic.mllib
 
+theories/Coqlib.v
```
Comment (gmalecha) [inline _CoqProject:14]: This file seems to be from compcert. It can't be in this repository. If this pull request depends on it, then we need to find a different way to factor the pull request.

File: _CoqProject [hunk 3]
```diff
@@ -11,4 +11,6 @@ src/cvc4Solver.ml
...
 theories/Tactic.v
+theories/Integers.v
```
Comment (gmalecha) [inline _CoqProject:16]: This also seems like it should not belong in this PR. If we do this, then we are only supporting *this* `Integers` library which is not a good way to go.

File: src/solver.ml [hunk 4]
```diff
@@ -138,9 +139,89 @@ struct
...
       end)
 
-  let r_pkg = ["Coq";"Reals";"Rdefinitions"]
   let logic_pkg = ["Coq";"Init";"Logic"]
 
+  let bin_pkg = ["Coq";"Numbers";"BinNums"]
+  let z_pkg = ["Coq";"ZArith";"BinInt";"Z"]
+
+  let int_pkg = ["SMTC";"Integers"]
+  let int64_pkg = ["SMTC";"Integers";"Int64"]
```
Comment (gmalecha) [inline src/solver.ml]: Coupling this plugin too closely to this `Integers` library.

File: src/solver.ml [hunk 5]
```diff
@@ -334,20 +669,22 @@ struct
...
+         Nat -> Format.fprintf out "(declare-const x%d %a)(assert (>= x%d 0))" k print_type v k
+        | _ -> Format.fprintf out "(declare-const x%d %a)" k print_type v)
 
   let print_a_string out s =
     Format.fprintf out "%s" s
 
   let conclusion_name = "__CONCLUSION__"
 
-  let write_instance ?pretty:(pretty=false) out inst =
+  let write_instance ?pretty:(pretty=true) out inst =
```
Comment (gmalecha) [inline src/solver.ml]: Why make pretty the default?

File: test-suite/Test.v [hunk 6]
```diff
@@ -1,11 +1,93 @@
 Require Import SMTC.Tactic.
```
Comment (gmalecha) [inline test-suite/Test.v:1]: The new tests should probably be in a new file.

**Review by gmalecha — CHANGES_REQUESTED** [review]
We can't include the CompCert library in this package, so we're going to need another way to do this.

Can you think of some way to refactor this to allow users to declare the way to convert Coq expressions into SMT expressions?

**Comment by gmalecha** [comment]
I noticed that a lot of what you did was build ocaml representations of the terms. Do you think we could do something like:

```
| Fn of string * expr list
```

And then support a way to map Gallina functions to strings? That would allow us to avoid the coupling between this code and libraries and make it possible for other people to extend the system without writing Ocaml code.

**Comment by wangjwchn** [comment]
I think I may need to clear up the commit then submit the PR.
